vignettes/Treatment.Rmd

```diff
@@ -15,50 +15,126 @@ knitr::opts_chunk$set(
+
+Prior to the simulation, the function `set_equilibrium` must also be used to generate equilibrium values for the human and mosquito populations.
+
+## Parameterisation and simulation
 
 ```{r}
+# Daily simulation timesteps for two years
 year <- 365
-sim_length <- 2 * year
+sim_length <- 1 * year
+
+# With a population size of 1000
 human_population <- 1000
 
-simparams <- get_parameters(list(human_population = human_population))
-simparams <- set_drugs(simparams, list(AL_params, DHA_PQP_params))
-```
+# Set human population size
+simparams <- get_parameters(overrides = list(human_population = human_population))
 
-Run the model for some starting EIRs and fts:
+# Update parameter set with chosen drug-specific parameters (AL and DHA/PQP)
+drug_params <- set_drugs(simparams, list(AL_params, DHA_PQP_params))
 
-```{r}
-outputs <- list()
-starting_EIR <- 10
-fts <- c(0, .25, .5, .75, 1.)
-
-for (ft in fts) {
-  simparams <- set_clinical_treatment(simparams, 1, 1, ft / 2)
-  simparams <- set_clinical_treatment(simparams, 2, 1, ft / 2)
-  simparams <- set_equilibrium(simparams, starting_EIR)
-  output <- run_simulation(sim_length, simparams)
-  # cut out the warmup
-  output <- output[output$timestep > year,]
-  outputs[[length(outputs) + 1]] <- output
-}
+# Choose initial EIR to 10
+starting_EIR <- 10 
+
+# Set treatment program for AL (drug index = 1)
+treatment_params <- set_clinical_treatment(
+  parameters = drug_params,
+  drug = 1,
+  timesteps =  c(100,300), # Treatment coverage changes on day 100 and day 200
+  coverages =  c(0.4,0)) # The initial treatment coverage (0%) is the default and does not need to be set
+
+# Set treatment program for DHA-PQP (drug index = 2)
+treatment_params <- set_clinical_treatment(
+  parameters = treatment_params,
+  drug =  2,
+  timesteps = c(200,300),
+  coverages =  c(0.3,0)) # Coverage as found in covs vector
+
+# Use set_equilibrium to update the parameter set for a given initial EIR
+treatment_params <- set_equilibrium(treatment_params, starting_EIR)
+
+# Run simulation:
+output <- run_simulation(sim_length, treatment_params)
+  
 ```
 
-Let's plot the effect of ft on PfPR2-10:
+## Visualisation
 
-```{r}
-df <- do.call('rbind', lapply(seq_along(outputs), function(i) {
-  df <- outputs[[i]][c('timestep', 'n_detect_730_3650', 'n_730_3650')]
-  df$ft <- fts[[i]]
-  df
-}))
-    
-ggplot(df) + geom_line(
-  aes(x = timestep, y = (n_detect_730_3650/n_730_3650), colour = ft)) + 
-  labs(x = "timestep", y = "PfPR2-10")
+Following simulation of malaria transmission under this treatment regimen, we can now visualise the effect of the regimen on *Pf*PR~2-10~ through time using the `n_detect_730_3650` and `n_730_3650` outputs.
+
+```{r, fig.align = 'center', fig.height = 5, fig.width = 7}
+# Create a new variable for Pf Prevalence (Age 2-10)
+output$PfPR2_10 <- output$n_detect_730_3650/output$n_730_3650
+
+# Plot results
+plot(x = output$timestep, y = output$PfPR2_10, type = "l",
```

Comment:
Can you add colored vertical blocks to indicate the times when there is AL being given (100-200) and when there is DHA_PQP being given (200-300)? Will just make it clearer that prevalence falls under both treatment options and then starts to rise again after 300 days. 

Cases also might show this change better than pfpr

Reply:
I've switched to n_detect_730_3650, extended the simulation to 2 years, and made both treatments regimens from day 300-600. I think the effect is a bit clearer now.